### DISCUSSION TITLE
Sort objects by period

### DISCUSSION THREAD

**Discussion by riquedev** [abhiabhi94/dj-hitcount#35] [Q&A]
I'm having trouble sorting the objects with the highest hits in a given period, I can't come up with a way to do it.
Currently I filter only the most "viewed" of the entire period, I use the syntax below:

```python
ctype = ContentType.objects.get_for_model(News)
HitCount = get_hitcount_model()
hit_count = HitCount.objects.filter(content_type=ctype).order_by('-hits').values_list('object_pk')
popular_news = News.objects.filter(pk__in=hit_count)
```

**Comment by abhiabhi94** [comment]
There is a method [`hits_in_last`](https://github.com/abhiabhi94/dj-hitcount/blob/main/hitcount/models.py#L53) for the `HitCount` model. I think if you can find a way to use this as a way to `annotate` the `hitcount` objects, you should be able to achieve what you want.

Please let me know if this doesn't work or you happen to figure out a better way.
